qe_metrics/libs/database.py

```
@@ -1,20 +1,115 @@
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        if self.connection:
+            self.connection.disconnect()
+            self.logger.info("Disconnected from the database")
 
-        # TODO: Add database connection
+    def bind_local_db_connection(self) -> None:
+        """
+        Bind a local database connection using SQLite.
+        """
+        sqlite_filepath = self.db_creds.get("local_filepath", DEFAULT_SQLITE_DB_FILEPATH)
+        self.logger.info(f"Local database connection enabled, using SQLite database at {sqlite_filepath}")
+        self.connection.bind(provider="sqlite", filename=sqlite_filepath, create_db=not os.path.exists(sqlite_filepath))
+
+    def bind_remote_db_connection(self) -> None:
+        """
+        Bind a remote database connection.
+        """
+        self.logger.info("Remote database connection enabled, connecting to database")
+        if self.remote_db_creds_exist():
+            self.connection.bind(
+                host=self.db_creds["host"],
+                user=self.db_creds["user"],
+                password=self.db_creds["password"],
+                database=self.db_creds["database"],
+                port=self.db_creds.get("port", DEFAULT_REMOTE_SQL_PORT),
+                provider=self.db_creds.get("provider", DEFAULT_REMOTE_SQL_PROVIDER),
+            )
+        else:
+            self.logger.error(
+                f'Some or all configuration values in the "database" section of {self.creds_file} are missing.'
+            )
+            raise click.Abort()
 
+    def remote_db_creds_exist(self) -> bool:
+        """
+        Check if the remote database values exist in the configuration file.
+
+        Returns:
+            bool: True if the remote database values exist, False otherwise.
+        """
+        required_keys = ["host", "user", "password", "database"]
+        return all(self.db_creds.get(key) is not None for key in required_keys)
```

please raise and log which keys are missing